### ISSUE TITLE
spack fails to build samrai 3.11.2

### ISSUE THREAD

**Issue by twang15** [spack/spack#8491]
While building samrai on LLNL quartz for version 3.11.2 with Intel compiler, spack expects there is include/boost/tr1 sub-directory, but the boost library installed by spack does not have such a directory, and the build fails.

### Expected Result

The build should succeed without an error.

### Actual Result

Spack fails at the configuration step, because it cannot find include/boost/tr1 directory for boost.

Error message:
     113    checking whether HDF5 link works... yes
     114    checking for BOOST installation...
  >> 115    configure: error: BOOST not found in /usr/workspace/wsa/wang102/spack/opt/spack/linux-rhel7-x86_64/intel-18.0.2/boost-1.67.0-5356zjcg5ezmk76eybujazxnfuz6wtql

### Steps to reproduce the issue
$ spack install samrai@3.11.2 %intel@18.0.2

### Other information
spack: 0.11.2, development branch
machine architecture: Intel(R) Xeon(R) CPU E5-2695 v4 @ 2.10GHz
OS: (Red Hat Enterprise Linux Server release 7.4 (Maipo)
compiler: Intel 18.0.2

$ spack spec samrai@3.11.2 %intel@18.0.2

Input spec
--------------------------------
samrai@3.11.2%intel@18.0.2

Concretized
--------------------------------
samrai@3.11.2%intel@18.0.2~debug arch=linux-rhel7-x86_64
    ^boost@1.67.0%intel@18.0.2+atomic+chrono~clanglibcpp cxxstd=default +date_time~debug+exception+filesystem~graph~icu+iostreams+locale+log+math~mpi+multithreaded~numpy patches=2ab6c72d03dec6a4ae20220a9dfd5c8c572c5294252155b85c6874d97c323199 +program_options~python+random+regex+serialization+shared+signals~singlethreaded+system~taggedlayout+test+thread+timer~versionedlayout+wave arch=linux-rhel7-x86_64
        ^bzip2@1.0.6%intel@18.0.2+shared arch=linux-rhel7-x86_64
        ^zlib@1.2.11%intel@18.0.2+optimize+pic+shared arch=linux-rhel7-x86_64
    ^hdf5@1.10.2%intel@18.0.2~cxx~debug~fortran~hl+mpi+pic+shared~szip~threadsafe arch=linux-rhel7-x86_64
        ^openmpi@3.1.0%intel@18.0.2~cuda fabrics=verbs ~java~memchecker~pmi schedulers= ~sqlite3~thread_multiple+vt arch=linux-rhel7-x86_64
            ^hwloc@1.11.9%intel@18.0.2~cairo~cuda+libxml2+pci+shared arch=linux-rhel7-x86_64
                ^libpciaccess@0.13.5%intel@18.0.2 arch=linux-rhel7-x86_64
                    ^libtool@2.4.6%intel@18.0.2 arch=linux-rhel7-x86_64
                        ^m4@1.4.18%intel@18.0.2 patches=3877ab548f88597ab2327a2230ee048d2d07ace1062efe81fc92e91b7f39cd00 +sigsegv arch=linux-rhel7-x86_64
                            ^libsigsegv@2.11%intel@18.0.2 arch=linux-rhel7-x86_64
                    ^pkgconf@1.4.2%intel@18.0.2 arch=linux-rhel7-x86_64
                    ^util-macros@1.19.1%intel@18.0.2 arch=linux-rhel7-x86_64
                ^libxml2@2.9.8%intel@18.0.2~python arch=linux-rhel7-x86_64
                    ^xz@5.2.4%intel@18.0.2 arch=linux-rhel7-x86_64
                ^numactl@2.0.11%intel@18.0.2 arch=linux-rhel7-x86_64
                    ^autoconf@2.69%intel@18.0.2 arch=linux-rhel7-x86_64
                        ^perl@5.26.2%intel@18.0.2+cpanm patches=0eac10ed90aeb0459ad8851f88081d439a4e41978e586ec743069e8b059370ac +shared+threads arch=linux-rhel7-x86_64
                            ^gdbm@1.14.1%intel@18.0.2 arch=linux-rhel7-x86_64
                                ^readline@7.0%intel@18.0.2 arch=linux-rhel7-x86_64
                                    ^ncurses@6.1%intel@18.0.2~symlinks~termlib arch=linux-rhel7-x86_64
                    ^automake@1.16.1%intel@18.0.2 arch=linux-rhel7-x86_64

**Comment by jrood-nrel** [comment]
Hmm. I'm seeing this as well. I will look into this.

**Comment by adamjstewart** [comment]
Is there a boost variant that needs to be enabled in the dependency?

**Comment by jrood-nrel** [comment]
I created #8494 to solve this. The latest version of SAMRAI 3.12.0 doesn't even use boost, so I would just use that @twang15 . Although I did fix v3.11.x by forcing an older version of boost. When using a newer boost, SAMRAI just couldn't find it. I don't know what version of boost that happened exactly, but I just chose boost 1.60.0 for SAMRAI 3.11.x. I tested 3.11.2, 3.11.5 and 3.12.0 with my pull request.

**Comment by twang15** [comment]
I also have a fix for samrai up to version 3.12.0.
And I figure out that boost 1.64.0 is the highest good version for samrai
version earlier than 3.12.0.

Shall we merge our fixes?

**Comment by adamjstewart** [comment]
@twang15 #8494 was already merged, but feel free to open up another PR to add in your changes as well.

**Comment by twang15** [comment]
Could anyone share experiences with me how to make PR pass spack test?

For Unit test, spack test gives me the following error:

==> Error: invalid literal for int() with base 10: 'unknown'



For Documentation test, it fails with:

usage: sphinx-build [OPTIONS] -o <OUTPUT_PATH> <MODULE_PATH>
[EXCLUDE_PATTERN, ...]
sphinx-build: error: argument -d/--maxdepth: invalid int value:
'_build/doctrees'

Configuration error:
The configuration file (or one of the modules it imports) called sys.exit()
make: *** [html] Error 2

**Comment by twang15** [comment]
I also wonder how Jon's PR passes the test since samrai cannot compile on
Quartz.
Because the samrai 3.12.0 requires c++11 flags, but there is no such flag
in the latest version.

**Comment by adamjstewart** [comment]
> I followed spack contribution guide here

The guide should probably make things clear, but unless you are modifying Spack internals, you don't really have to worry about `spack test`. And unless you are modifying the documentation, you shouldn't have to worry about the documentation tests.

> For Unit test, spack test gives me the following error:

Can you give me more context? Which test is failing? What OS are you on?

> For Documentation test, it fails with:

This looks like it's related to https://github.com/rtfd/readthedocs.org/issues/3769. Apparently the Sphinx options have changed in someway. This isn't an issue for us as we lock the Sphinx version to 1.7.0 in our `requirements.txt`, but I'll investigate anyway because I would like to be able to use the latest version of Sphinx.

> I also wonder how Jon's PR passes the test since samrai cannot compile on Quartz.

Unfortunately, Spack's PR tests don't actually build anything at the moment, as many packages have so many dependencies that the build time would timeout Travis's time limit.

In summary, if you can get `samrai` to build and everything looks good to you, submit a PR and I'll take a look. Don't worry about the tests too much; if you're only modifying a package, `spack flake8` is the most likely thing to cause problems.

I am curious about these other test failures though. If I can reproduce them that will help greatly.

**Comment by adamjstewart** [comment]
I opened an issue with Sphinx to get to the bottom of this: https://github.com/sphinx-doc/sphinx/issues/5104

At first glance, I can't see what is causing the problem.

**Comment by twang15** [comment]
The test is performed on LLNL Quartz.

"spack test" is to perform the unit test described in the contribution
guide.
To reproduce the issue, please see the following steps.

1. git clone spack's and switch to develop branch on Quartz, if you have
access to it.
2. perform unit test according to the contribution guide.

The reported error "==> Error: invalid literal for int() with base 10:
'unknown'" is the only output "spack test" command.

For documentation test, you can reproduce the error by following the
contribution guide on spack's develop branch.


spack flake8 test succeeds, so I will create the PR.

**Comment by twang15** [comment]
@jrood-nrel @adamjstewart @tgamblin I create a PR (#8501). Compilation tests for 3.12.0, 3.11.3~3.11.5 on both ICC/GCC are good; flake8 test also passes.
